gzhttp/compress.go

```diff
@@ -464,6 +464,12 @@ func NewWrapper(opts ...option) (func(http.Handler) http.HandlerFunc, error) {
 	return func(h http.Handler) http.HandlerFunc {
 		return func(w http.ResponseWriter, r *http.Request) {
 			w.Header().Add(vary, acceptEncoding)
+			if contentGzip(r) {
+				if readerGzipBody, err := gzip.NewReader(r.Body); err == nil {
+					r.Body = io.NopCloser(readerGzipBody)
```

Comment:
The header field must be removed to avoid upstream confusion.

Reply:
done